lib/views/push-pull-view.js

```diff
@@ -2,37 +2,196 @@ import React from 'react';
 import PropTypes from 'prop-types';
 import cx from 'classnames';
 
+import {RemotePropType, BranchPropType} from '../prop-types';
+
+
+function getIconClass(icon, isSyncing = false) {
+  return cx(
+    'github-PushPull-icon',
+    'icon',
+    isSyncing ? 'icon-sync' : `icon-${icon}`,
+    {'animate-rotate': isSyncing},
+  );
+}
+
 export default class PushPullView extends React.Component {
   static propTypes = {
-    pushInProgress: PropTypes.bool,
-    fetchInProgress: PropTypes.bool,
+    currentBranch: BranchPropType.isRequired,
+    currentRemote: RemotePropType.isRequired,
+    isSyncing: PropTypes.bool,
     behindCount: PropTypes.number,
     aheadCount: PropTypes.number,
+    push: PropTypes.func.isRequired,
+    pull: PropTypes.func.isRequired,
+    fetch: PropTypes.func.isRequired,
+    originExists: PropTypes.bool,
   }
 
   static defaultProps = {
-    pushInProgress: false,
-    fetchInProgress: false,
+    isSyncing: false,
     behindCount: 0,
     aheadCount: 0,
   }
 
+
+  componentDidMount() {
+    this.setTooltip();
+  }
+
+  componentDidUpdate() {
+    this.tooltip.dispose();
+    this.setTooltip();
+  }
+
+  setTooltip = () => {
+    this.tooltip = atom.tooltips.add(this.tileNode, {
+      title: `<div style="text-align: left; line-height: 1.2em;">${this.tooltipTarget.dataset.tooltip}</div>`,
+      delay: {show: 200, hide: 100},
+      html: true,
+    });
+  }
+
+  onClickPush = clickEvent => {
+    if (this.props.isSyncing) {
+      return;
+    }
+    this.props.push({
+      force: clickEvent.metaKey || clickEvent.ctrlKey,
+      setUpstream: !this.props.currentRemote.isPresent(),
+    });
+  }
```

Comment:
Is there a reason these are:

```js
methodName = () => {
  //
}
```

Rather than:

```js
methodName() {
  //
}
```

? It isn't a pattern I've seen before.

Reply:
Since arrow functions don't change their context, `this` inside of these event handlers automatically refers to the component. So this pattern avoids having to do e.g. 
```JS
function constructor(props) {
    super(props);
    this.onClickPush = this.onClickPush.bind(this);
}
```
or 
```JSX
<a onClick={this.onClick.bind(this)}>Click me</a>
```
to keep the correct context in event handlers.

If introducing this new pattern is unwelcome, I'll be happy to replace it with on of the more traditional approaches.

Reply:
Oh! Neat. 👍 Elsewhere we're using the `@autobind` decorator, but honestly I kind of prefer this - it doesn't rely on is-it-or-isn't-it-deprecated Babel plugins :smile: